modules/nf-core/samtools/bgzip/main.nf

```diff
@@ -0,0 +1,55 @@
+process SAMTOOLS_BGZIP {
+    tag "$fasta"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.21--h50ea8bc_0' :
+        'biocontainers/samtools:1.21--h50ea8bc_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path ("*.bgzip.fa.gz") , emit: fa
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    FILE_TYPE=\$(htsfile $fasta)
+    case "\$FILE_TYPE" in
+        *BGZF-compressed*)
+            ln -s $fasta ${prefix}.bgzip.fa.gz ;;
+        *gzip-compressed*)
+            zcat  $fasta | bgzip -c $args -@${task.cpus} > ${prefix}.bgzip.fa.gz ;;
+        *bzip2-compressed*)
+            bzcat $fasta | bgzip -c $args -@${task.cpus} > ${prefix}.bgzip.fa.gz ;;
+        *XZ-compressed*)
+            xzcat $fasta | bgzip -c $args -@${task.cpus} > ${prefix}.bgzip.fa.gz ;;
+        *)
+            bgzip -c $args -@${task.cpus} $fasta > ${prefix}.bgzip.fa.gz ;;
```

Comment:
I'd rather we keep the extension short similarly to how we do in https://github.com/nf-core/modules/blob/master/modules/nf-core/tabix/bgzip/main.nf

ie only `.gz`

Reply:
Hi @maxulysse I have implemented the changes that you have requested.